shared/src/commonMain/kotlin/com/hub/media/ui/RestoreViewModel.kt

```
@@ -49,9 +59,26 @@ public class RestoreViewModel(
         _uiState.value = RestoreUiState.Validating
         viewModelScope.launch {
+            // Read now, while the live database still holds whatever key it holds -- see
+            // RestoreUiState.AwaitingConfirmation.apiKeyWillBeCleared's KDoc for why this can't
+            // be deferred to any later point in the restore flow. Presence only; the key's
+            // value is never read here or held by this ViewModel.
+            val keySet =
+                try {
+                    settingsRepository.getGoogleBooksApiKey() != null
+                } catch (e: CancellationException) {
+                    throw e
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)